### DISCUSSION TITLE
Snapcast on same server as HA?

### DISCUSSION THREAD

**Discussion by wyllys66** [snapcast/snapcast#1280] [Q&A] [answered]
Can snapcast server be installed on the same RPi5 as Homeassistant?  I'm trying to integrate a music player that will let me play the audio out through the HDMI audio output on the RPI instead of streaming over bluetooth or some other OTA protocol because I have an older Denon amplifier driving my built-in home speakers.

My thought was to put the snapcast server software on the same RPI as the HA and then use the HA (or Music Assistant) snapcast integration to control it from the HA interface.   Would this work?

**Comment by wyllys66** [comment, marked as answer]
I figured this out. The answer is NO (or not easily).

The HAOS is locked up pretty tight, and while I could perhaps hack my way around to get to the hypervisor level and install new packages, it seemed easier to just setup a snapserver and snapclient on the rpi5 and put the HA on a different server and point it at my standalone snapserver.

**Comment by milnivlek** [comment]
You can do it if you use Home Assistant Container instead of HAOS. That's how I have it set up -- Snapcast and Docker Compose on the same host, and Home Assistant running within Docker. Works perfectly.

**Comment by wyllys66** [comment]
Thanks!  I did end up doing something similar - I didn't want to overload the RPI with docker server and containers so I put the HA service in a container on a different server on my network and left the snapcast to run on the rpi connected to the amp.

**Comment by Bram-diederik** [comment]
a PI4 can do that, i have it running on a remote location. 
surely a PI5 can with her hands in her pockets. 
use default 64 bit setup. (with or with out desktop(you can always stop it from loading))
use the ethernet 
make a network bridge
install libvirt
setup bridge mode for libvirt
install home assistant haos on libvirt
https://www.home-assistant.io/installation/linux/
Note the arm img is not on the howto
use https://github.com/home-assistant/operating-system/releases (arch64 cqow)

**Comment by vednolacni** [comment]
https://github.com/Art-Ev/addon-snapserver
